sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobOutputStream.java

```diff
@@ -134,62 +139,67 @@ void commit() {
     }
 
     private static final class BlockBlobOutputStream extends BlobOutputStream {
-        private final BlobRequestConditions requestConditions;
-        private final String blockIdPrefix;
-        private final List<String> blockList;
-        private final BlockBlobAsyncClient client;
+
+        private FluxSink<ByteBuffer> sink;
+
+        boolean complete;
 
         private BlockBlobOutputStream(final BlockBlobAsyncClient client,
-            final BlobRequestConditions requestConditions) {
+            final ParallelTransferOptions parallelTransferOptions, final BlobHttpHeaders headers,
+            final Map<String, String> metadata, final AccessTier tier, final BlobRequestConditions requestConditions) {
             super(BlockBlobClient.MAX_STAGE_BLOCK_BYTES);
-            this.client = client;
-            this.requestConditions = (requestConditions == null) ? new BlobRequestConditions() : requestConditions;
-            this.blockIdPrefix = UUID.randomUUID().toString() + '-';
-            this.blockList = new ArrayList<>();
-        }
 
-        /**
-         * Generates a new block ID to be used for PutBlock.
-         *
-         * @return Base64 encoded block ID
-         */
-        private String getCurrentBlockId() {
-            String blockIdSuffix = String.format("%06d", this.blockList.size());
-            return Base64.getEncoder().encodeToString((this.blockIdPrefix + blockIdSuffix)
-                .getBytes(StandardCharsets.UTF_8));
-        }
+            BlobAsyncClient blobClient = prepareBuilder(client).buildAsyncClient();
 
-        private Mono<Void> writeBlock(Flux<ByteBuffer> blockData, String blockId, long writeLength) {
-            return client.stageBlockWithResponse(blockId, blockData, writeLength, null,
-                this.requestConditions.getLeaseId())
-                .then()
-                .onErrorResume(BlobStorageException.class, e -> {
-                    this.lastError = new IOException(e);
-                    return Mono.empty();
-                });
+            Flux<ByteBuffer> fbb = Flux.create((FluxSink<ByteBuffer> sink) -> this.sink = sink);
+            fbb.subscribe(); // Subscribe by upload takes too long
```

Comment:
Can you expound on this comment--We need to subscribe so that the sink is actually created and because this subscriber doesn't do anything and no data has started flowing, there are no drawbacks to this extra subscribe.